src-v2/NTS.Domain.Core/Objects/Startlist.cs

```diff
@@ -5,49 +5,38 @@ namespace NTS.Domain.Core.Objects;
 
 public class StartList
 {
-    public StartList() { }
+    static readonly TimeSpan START_EXPIRY_TIME = TimeSpan.FromMinutes(15);
 
-    public List<Start> Upcoming { get; set; } = new List<Start>();
-    public List<Start> History { get; set; } = new List<Start>();
+    public StartList() { }
 
-    public void OrderHistoryByAscending()
-    {
-        History = History.OrderBy(s => s.StartAt).ToList();
-    }
+    public List<Start> Starts { get; set; } = [];
 
-    public void Replace(Start start)
+    public List<Start> History
     {
-        if (History.Contains(start))
+        get
         {
-            History.Remove(start);
-            History.Add(start);
-            OrderHistoryByAscending();
+            var now = DateTime.Now.TimeOfDay;
+            return Starts
+                .Where(s => now - s.StartAt > START_EXPIRY_TIME)
+                .OrderBy(s => s.StartAt)
+                .ToList();
         }
-        if (Upcoming.Contains(start))
-        {
-            Upcoming.Remove(start);
-            Upcoming.Add(start);
-        }
-    }
-
-    public void Add(Start start)
-    {
-        Upcoming.Add(start);
     }
 
-    public void Expire(Start start)
+    public List<Start> Upcoming
     {
-        if (!History.Contains(start))
+        get
         {
-            Upcoming.Remove(start);
-            History.Add(start);
-            OrderHistoryByAscending();
+            var now = DateTime.Now.TimeOfDay;
+            return Starts
+                .Where(s => now - s.StartAt <= START_EXPIRY_TIME)
+                .OrderBy(s => s.StartAt)
+                .ToList();
```

Comment:
Once again, this is a copy-paste. You have to start to put more thought in reusing code. In this case, it's a textbook example, because when you change the type of `StartAt` you'll have to make changes to both properties, rather than a single method.